action/candidate_register.go

```diff
@@ -293,8 +292,8 @@ func (cr *CandidateRegister) SanityCheck() error {
 	if cr.Amount().Sign() <= 0 {
 		return errors.Wrap(ErrInvalidAmount, "negative value")
 	}
-	if err := validator.ValidateCandidateNameForStake2(cr.Name()); err != nil {
-		return errors.Wrap(ErrInvalidCandidateName, err.Error())
+	if err := isValidCandidateName(cr.Name()); err != nil {
+		return err
```

Comment:
1. staking protocol can use this, delete `isValidCandidateName` inside \staking
2. ioctl can use this